kubemarine/procedures/check_iaas.py

```diff
@@ -1240,6 +1241,127 @@ def vips_connectivity(cluster: KubernetesCluster) -> None:
+    failed_nodes: Set[str] = set()
+    recv_cmd: Dict[str, str] = {}
+    trns_cmd: Dict[str, str] = {}
+
+    binary_check_path = utils.get_internal_resource_path('./resources/scripts/ipip_check')
+    ipip_check = "/tmp/%s" % uuid.uuid4().hex
+    recv_log = "/tmp/%s" % uuid.uuid4().hex
+    # Random message
+    random.seed()
+    msg = ''.join(random.choices(string.ascii_letters + string.digits, k=15))
+    # Random IP from class E
+    int_ip = random.randint(4026531841, 4294967294)
+    fake_addr = str(ipaddress.IPv4Address(int_ip))
+    # That is used as number of packets for transmitter
+    timeout = int(cluster.inventory['globals']['timeout_download'])
+    for node in group.get_ordered_members_configs_list():
+        host = node['internal_address']
+        # Transmitter start command
+        # Transmitter starts first and sends IPIP packets every 1 second until the timeout comes or
+        # the process is killed by terminating command
+        trns_item_cmd: List[str] = []
+        for node_item in group.get_ordered_members_configs_list():
+            if node_item['internal_address'] != host:
+                trns_item_cmd.append(f"nohup {ipip_check} -mode client -src {host} -int {fake_addr} "
+                                     f"-ext {node_item['internal_address']} -dport {random_port} "
+                                     f"-msg {msg} -timeout {timeout} > /dev/null 2>&1 & echo $! >> {recv_log}.pid")
+        trns_cmd[host] = '& sudo '.join(trns_item_cmd)
+        # Receiver start command
+        # Receiver starts after the transmitter and try to get IPIP packets within 3 seconds from eache node
+        recv_cmd[host] = f"{ipip_check} -mode server -ext {host} -int {fake_addr} -dport {random_port} " \
+                         f"-msg {msg} -timeout 3 > {recv_log}.log 2>{recv_log}.err"
+
+    try:
+        collector = CollectorCallback(group.cluster)
+        cluster.log.debug("Copy binaries to the nodes")
+        group.put(binary_check_path, ipip_check)
+        group.sudo(f"sudo chmod +x {ipip_check}")
+        # Run transmitters
+        cluster.log.debug("Run transmitters")
+        with group.new_executor() as exe:
+            for node_exe in exe.group.get_ordered_members_list():
+                host_int = node_exe.get_config()['internal_address']
+                node_exe.sudo(f"{trns_cmd[host_int]}")
+        # Run receivers
+        cluster.log.debug("Run receivers")
+        with group.new_executor() as exe:
+            for node_exe in exe.group.get_ordered_members_list():
+                host_int = node_exe.get_config()['internal_address']
+                node_exe.sudo(f"{recv_cmd[host_int]}", warn=True)
+        # Get results from the log file
+        cluster.log.debug("Check logs")
+        with group.new_executor() as exe:
+            for node_exe in exe.group.get_ordered_members_list():
+                node_exe.sudo(f"cat {recv_log}.log", warn=True, callback=collector)
```

Comment:
When we run receivers, as I understand we are synchronously waiting for them to finish execution, only after that we collect results from log file, is it correct? In this case, why we can not just print results to stdout so that they are returned immediately after receivers finish execution? We will use one less SSH command

Reply:
Yes, sure